File: 백준/Silver/1817. 짐 챙기는 숌/짐 챙기는 숌.py
import sys
input = lambda: sys.stdin.readline().strip()

n, m = map(int, input().split())
if n == 0:
    print(0)
else:
    arr = list(map(int, input().split()))

    cnt = 0
    temp = 0
    for i in range(n):
        if temp + arr[i] < m:
            temp += arr[i]
        elif temp + arr[i] > m:
            cnt += 1
            temp = arr[i]
        else:
            cnt += 1
            temp = 0
    if temp > 0:
        print(cnt + 1)
    else:
        print(cnt)

File: 백준/Silver/1817. 짐 챙기는 숌/짐 챙기는 숌.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

	public static void main(String[] args) throws IOException {
		
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = null;
		st = new StringTokenizer(br.readLine());
		
		int n = Integer.parseInt(st.nextToken());
		int m = Integer.parseInt(st.nextToken());
		
		if(n == 0)
			System.out.println(0);
		
		else {
			int[] arr = new int[n];
			st = new StringTokenizer(br.readLine());
			for(int i = 0; i < n;i++) {
				arr[i] = Integer.parseInt(st.nextToken());
			}
			
			int cnt = 0;
			int temp = 0;
			for(int i = 0; i < n; i++) {
				if(temp + arr[i] < m)
					temp += arr[i];
				else if (temp + arr[i] > m) {
					cnt += 1;
					temp = arr[i];
				}
				else {
					cnt += 1;
					temp = 0;
				}
					
			}
			
			if (temp > 0)
				System.out.println(cnt + 1);
			else
				System.out.println(cnt);
			
			
		}
		

	}

}
